### ISSUE TITLE
pipeline lock block doesn't get requested resource when reservation is lifted

### ISSUE THREAD

**Issue by glance-** [jenkinsci/lockable-resources-plugin#86]
A resource is reserved.

A pipeline job is waiting for that resource with a ```lock("resource") {}...``` block.

When the resource is unreserved, the pipeline job still waits for the resource, and never gets it.

**Comment by dabrowt1** [comment]
I'm also observing this. This happens even if running all tests on a single node. I'm using parallel tests if that is important:

```
    stage("Tests") {
      steps {
        parallel(
            "DB": {
              lock('db') {              
                ws(WS_DB) {
                  checkout scm            
                  sh './tests-db'
                }
              }
            },
            "Mongo": {
              lock('mongo') {
                ws(WS_MONGO) {
                  checkout scm
                  sh './tests-mongo'
                }
              }
            },
...
```

**Comment by dcendents** [comment]
You could always give a try to my PR #87, when testing it I noticed that bug and also fixed it (call the same method that is called when a lock is released by a job).

**Comment by DJViking** [comment]
When I lock the resource my self manually, then the pipeline build never acquires the resource after I release it. However if I start two builds of same pipeline, build one acquires the resource locks and build two waits. When build one releases the resource locks, then build two acquires the locks. So it seems only a problem when locking the resources manually.

My problem however seems to be when using lock on label. It might be related to this issue.
https://groups.google.com/forum/#!topic/jenkinsci-users/QiuXKcadHZw

I have defined 3 resources, all with the same label.
Pipeline build ONE has acquired a lock on label, thus locking all three resources.
Pipeline build TWO is waiting for a resource within each of the 3 parallell stages.
When Pipeline build ONE has released its lock then the Pipeline build TWO only manages to acquire the first resource of the 3 resources. The entire builds hangs waiting for the remaining two resources which are free.

**Comment by amuniz** [comment]
> Will PR #87 fix my problem?

No, it will be fixed by https://github.com/jenkinsci/lockable-resources-plugin/pull/70

**Comment by DJViking** [comment]
Isn't the #70 fix just for manual reservation/unreservation of a resource by a user?
Will it fix my issue with locking on label and resource by two consecutive builds?

**Comment by TobiX** [comment]
This should have been fixed in release 2.3
